chemprop/utils/mixins.py

```diff
@@ -4,17 +4,47 @@
 
 class ReprMixin:
     def __repr__(self) -> str:
-        items = self.get_params()
+        default_params = self._get_default_params()
+        param_repr_list = [
+            f"{name}={repr(value)}"
+            for name, value in self.get_params(deep=False).items()
+            if value != default_params[name]
+        ]
 
-        if len(items) > 0:
-            keys, values = zip(*items)
-            sig = inspect.signature(self.__class__)
-            defaults = [sig.parameters[k].default for k in keys]
-            items = [(k, v) for k, v, d in zip(keys, values, defaults) if v != d]
+        return f"{self.__class__.__name__}({','.join(param_repr_list)})"
 
-        argspec = ", ".join(f"{k}={repr(v)}" for k, v in items)
+    @classmethod
+    def _get_default_params(cls) -> dict[str, Any]:
+        """Get the default parameters for initializing the class."""
+        sig = inspect.signature(cls)
+        return {k: v.default for k, v in sig.parameters.items()}
 
-        return f"{self.__class__.__name__}({argspec})"
+    def _get_param_names(self) -> list[str]:
+        """Get the parameter names for initializing the class."""
+        return sorted(self._get_default_params().keys())
 
-    def get_params(self) -> Collection[tuple[str, Any]]:
-        return self.__dict__.items()
+    def get_params(self, deep: bool = True) -> dict[str, Any]:
+        """Get the current parameters of the class, required to initialize the class.
+
+        This method mimics the behavior of :method:`sklearn.base.BaseEstimator.get_params`.
+
+        Parameters
+        ----------
+        deep : bool, default=True
+            If True, will return the parameters for this estimator and
+            contained subobjects that are estimators.
+
+        Returns
+        -------
+        dict[str, Any]
+            Parameter names mapped to their values.
```

Comment:
can this be more descriptive? Something like:
```suggestion
            The parameter names and respective values used to initialize this class. That is, an identical initialization of this object can be achieved via:
            
            ```
            o: ReprMixin
            o_copy = type(o)(**o.get_params())
            ```
```

Reply:
though I don't think this is correct in the case of `deep=True`...

Reply:
Updated the docstring